addon/components/paper-switch.js

```diff
@@ -74,20 +74,21 @@ export default BaseFocusable.extend(RippleMixin, ProxiableMixin, ColorMixin, {
   },
 
   _setupSwitch() {
-    this.set('switchWidth', this.$('.md-bar').width());
+    this.set('switchWidth', this.$('.md-thumb-container').innerWidth());
 
-    // Enable dragging the switch
     let switchContainer = this.$('.md-container').get(0);
-    let switchContainerHammer = new Hammer(switchContainer);
-    this._switchContainerHammer = switchContainerHammer;
-    switchContainerHammer.get('pan').set({ threshold: 1 });
-    switchContainerHammer.on('panstart', run.bind(this, this._dragStart));
-    switchContainerHammer.on('panmove', run.bind(this, this._drag));
-    switchContainerHammer.on('panend', run.bind(this, this._dragEnd));
-
-    let switchHammer = new Hammer(this.element);
-    this._switchHammer = switchHammer;
-    switchHammer.on('tap', run.bind(this, this._dragEnd));
+    let switchHammer = new Hammer(switchContainer);
+    this._switchContainerHammer = switchHammer;
+
+    // Enable dragging the switch container
+    switchHammer.get('pan').set({ threshold: 1 });
+    switchHammer.on('panstart', run.bind(this, this._dragStart))
```

Comment:
Run.bind wraps handlers so that the handler has the correct context(i.e. this). If you look at the body of each handlers we're changing the state of the component through this.set invocation